### PR TITLE
chore: update test-all-versions configs to test fewer package versions

### PR DESCRIPTION
**Pull request by trentm**
The TAV tests in CI are taking 2.5h - 3h, for a total usage of over 13h
compute time. E.g.: https://github.com/open-telemetry/opentelemetry-js-contrib/actions/runs/11233652488/usage

This change reduces the number of tested versions from 839 to 317
(for a TAV run with Node.js v20). For a few of the .tav.yml files the
tested version *range* was actually increased to capture some versions
that the instrumentation supported, but were not being tested by TAV.

### CODE REVIEW

**Comment by trentm** [comment]
# raw data of number of versions run before and after

For Node.js v20.
This is captured by running:

```
npm run --if-present --workspaces test-all-versions -- --verbose --dry-run | \rg '(test-all-versions|matching)'
```

<details>
<summary>before</summary>

```
> @opentelemetry/instrumentation-amqplib@0.42.0 test-all-versions
-- 13 package versions matching filter: 0.5.5, 0.5.6, 0.6.0, 0.7.0, 0.7.1, 0.8.0, 0.9.0, 0.9.1, 0.10.0, 0.10.1, 0.10.2, 0.10.3, 0.10.4
> @opentelemetry/instrumentation-cucumber@0.9.0 test-all-versions
-- 14 package versions matching filter: 10.0.0, 10.0.1, 10.1.0, 10.2.1, 10.3.0, 10.3.1, 10.3.2, 10.4.0, 10.5.0, 10.5.1, 10.6.0, 10.7.0, 10.8.0, 10.9.0
-- 33 package versions matching filter: 8.0.0, 8.1.0, 8.1.1, 8.1.2, 8.2.0, 8.2.1, 8.2.2, 8.3.0, 8.3.1, 8.4.0, 8.5.0, 8.5.1, 8.5.2, 8.5.3, 8.6.0, 8.7.0, 8.8.0, 8.9.0, 8.9.1, 8.10.0, 8.11.0, 8.11.1, 9.0.0, 9.0.1, 9.1.0, 9.1.1, 9.1.2, 9.2.0, 9.3.0, 9.4.0, 9.5.0, 9.5.1, 9.6.0
> @opentelemetry/instrumentation-dataloader@0.12.0 test-all-versions
-- 5 package versions matching filter: 2.0.0, 2.1.0, 2.2.0, 2.2.1, 2.2.2
> @opentelemetry/instrumentation-lru-memoizer@0.40.0 test-all-versions
-- 24 package versions matching filter: 1.3.0, 1.3.1, 1.4.0, 1.5.0, 1.6.0, 1.7.0, 1.8.0, 1.9.0, 1.10.0, 1.10.1, 1.11.0, 1.11.1, 1.11.2, 1.12.0, 1.13.0, 2.0.0, 2.0.1, 2.1.0, 2.1.1, 2.1.2, 2.1.3, 2.1.4, 2.2.0, 2.3.0
> @opentelemetry/instrumentation-mongoose@0.42.0 test-all-versions
-- 8 package versions matching filter: 8.0.4, 8.1.3, 8.2.4, 8.3.5, 8.4.5, 8.5.5, 8.6.4, 8.7.0
-- 9 package versions matching filter: 7.0.5, 7.1.2, 7.2.4, 7.3.4, 7.4.5, 7.5.4, 7.6.13, 7.7.0, 7.8.2
-- 19 package versions matching filter: 5.9.29, 5.10.19, 5.11.20, 5.12.15, 5.13.22, 6.0.15, 6.1.10, 6.2.11, 6.3.9, 6.4.7, 6.5.5, 6.6.7, 6.7.5, 6.8.4, 6.9.3, 6.10.5, 6.11.6, 6.12.9, 6.13.3
> @opentelemetry/instrumentation-socket.io@0.42.0 test-all-versions
-- 8 package versions matching filter: 4.5.3, 4.6.2, 4.7.1, 4.7.2, 4.7.3, 4.7.4, 4.7.5, 4.8.0
-- 1 package versions matching filter: 3.1.2
-- 6 package versions matching filter: 2.1.1, 2.2.0, 2.3.0, 2.4.0, 2.5.0, 2.5.1
> @opentelemetry/instrumentation-tedious@0.14.0 test-all-versions
-- 1 package versions matching filter: 19.0.0
-- 1 package versions matching filter: 18.6.2
-- 1 package versions matching filter: 17.0.0
-- 5 package versions matching filter: 12.3.0, 13.2.0, 14.7.0, 15.1.3, 16.7.1
-- 11 package versions matching filter: 1.15.0, 2.7.1, 3.0.1, 4.2.0, 5.0.3, 6.7.1, 7.0.0, 8.3.1, 9.2.3, 10.0.0, 11.8.0
> @opentelemetry/instrumentation-undici@0.6.0 test-all-versions
-- 7 package versions matching filter: 6.0.0, 6.4.0, 6.9.0, 6.13.0, 6.18.1, 6.19.6, 6.19.8
-- 7 package versions matching filter: 5.12.0, 5.16.0, 5.21.0, 5.25.1, 5.26.3, 5.28.2, 5.28.4
> @opentelemetry/instrumentation-aws-sdk@0.44.0 test-all-versions
-- 76 package versions matching filter: 3.24.0, 3.94.0, 3.202.0, 3.296.0, 3.388.0, 3.496.0, 3.503.1, 3.504.0, 3.507.0, 3.509.0, 3.511.0, 3.513.0, 3.514.0, 3.515.0, 3.521.0, 3.523.0, 3.525.0, 3.529.0, 3.529.1, 3.533.0, 3.535.0, 3.536.0, 3.540.0, 3.547.0, 3.549.0, 3.552.0, 3.554.0, 3.556.0, 3.563.0, 3.564.0, 3.565.0, 3.567.0, 3.568.0, 3.569.0, 3.572.0, 3.574.0, 3.575.0, 3.576.0, 3.577.0, 3.582.0, 3.583.0, 3.587.0, 3.588.0, 3.590.0, 3.592.0, 3.596.0, 3.598.0, 3.600.0, 3.606.0, 3.609.0, 3.613.0, 3.614.0, 3.616.0, 3.620.0, 3.620.1, 3.621.0, 3.622.0, 3.623.0, 3.624.0, 3.629.0, 3.631.0, 3.632.0, 3.635.0, 3.637.0, 3.645.0, 3.649.0, 3.650.0, 3.651.1, 3.654.0, 3.658.0, 3.658.1, 3.662.0, 3.664.0, 3.665.0, 3.666.0, 3.667.0
-- 75 package versions matching filter: 3.6.1, 3.55.0, 3.180.0, 3.289.0, 3.385.0, 3.498.0, 3.529.1, 3.533.0, 3.534.0, 3.535.0, 3.536.0, 3.537.0, 3.540.0, 3.549.0, 3.550.0, 3.552.0, 3.554.0, 3.556.0, 3.563.0, 3.564.0, 3.565.0, 3.567.0, 3.568.0, 3.569.0, 3.572.0, 3.574.0, 3.575.0, 3.576.0, 3.577.0, 3.582.0, 3.583.0, 3.587.0, 3.588.0, 3.590.0, 3.591.0, 3.592.0, 3.596.0, 3.598.0, 3.600.0, 3.606.0, 3.608.0, 3.609.0, 3.613.0, 3.614.0, 3.616.0, 3.617.0, 3.620.0, 3.620.1, 3.621.0, 3.622.0, 3.623.0, 3.624.0, 3.626.0, 3.627.0, 3.629.0, 3.631.0, 3.632.0, 3.633.0, 3.635.0, 3.637.0, 3.645.0, 3.649.0, 3.650.0, 3.651.0, 3.651.1, 3.654.0, 3.657.0, 3.658.0, 3.658.1, 3.662.0, 3.663.0, 3.664.0, 3.665.0, 3.666.0, 3.667.0
-- 150 package versions matching filter: 2.308.0, 2.556.0, 2.801.0, 2.1049.0, 2.1297.0, 2.1546.0, 2.1548.0, 2.1549.0, 2.1550.0, 2.1551.0, 2.1552.0, 2.1553.0, 2.1554.0, 2.1555.0, 2.1556.0, 2.1557.0, 2.1558.0, 2.1559.0, 2.1560.0, 2.1561.0, 2.1562.0, 2.1563.0, 2.1564.0, 2.1565.0, 2.1566.0, 2.1567.0, 2.1568.0, 2.1569.0, 2.1570.0, 2.1571.0, 2.1572.0, 2.1573.0, 2.1574.0, 2.1575.0, 2.1576.0, 2.1577.0, 2.1578.0, 2.1579.0, 2.1580.0, 2.1581.0, 2.1582.0, 2.1583.0, 2.1584.0, 2.1585.0, 2.1586.0, 2.1587.0, 2.1588.0, 2.1589.0, 2.1590.0, 2.1591.0, 2.1592.0, 2.1593.0, 2.1594.0, 2.1595.0, 2.1596.0, 2.1597.0, 2.1598.0, 2.1599.0, 2.1600.0, 2.1601.0, 2.1602.0, 2.1603.0, 2.1604.0, 2.1605.0, 2.1606.0, 2.1607.0, 2.1608.0, 2.1609.0, 2.1610.0, 2.1611.0, 2.1612.0, 2.1613.0, 2.1614.0, 2.1615.0, 2.1616.0, 2.1617.0, 2.1618.0, 2.1619.0, 2.1620.0, 2.1621.0, 2.1622.0, 2.1623.0, 2.1624.0, 2.1625.0, 2.1626.0, 2.1627.0, 2.1628.0, 2.1629.0, 2.1630.0, 2.1631.0, 2.1632.0, 2.1633.0, 2.1634.0, 2.1635.0, 2.1636.0, 2.1637.0, 2.1638.0, 2.1639.0, 2.1640.0, 2.1641.0, 2.1642.0, 2.1643.0, 2.1644.0, 2.1645.0, 2.1646.0, 2.1647.0, 2.1648.0, 2.1649.0, 2.1650.0, 2.1651.0, 2.1652.0, 2.1653.0, 2.1654.0, 2.1655.0, 2.1656.0, 2.1657.0, 2.1658.0, 2.1659.0, 2.1660.0, 2.1661.0, 2.1662.0, 2.1663.0, 2.1664.0, 2.1665.0, 2.1666.0, 2.1667.0, 2.1668.0, 2.1669.0, 2.1670.0, 2.1671.0, 2.1672.0, 2.1673.0, 2.1674.0, 2.1675.0, 2.1676.0, 2.1677.0, 2.1678.0, 2.1679.0, 2.1680.0, 2.1681.0, 2.1682.0, 2.1683.0, 2.1684.0, 2.1685.0, 2.1686.0, 2.1687.0, 2.1688.0, 2.1689.0, 2.1690.0, 2.1691.0
> @opentelemetry/instrumentation-bunyan@0.41.0 test-all-versions
-- 4 package versions matching filter: 1.8.12, 1.8.13, 1.8.14, 1.8.15
> @opentelemetry/instrumentation-express@0.43.0 test-all-versions
-- 17 package versions matching filter: 4.16.2, 4.16.3, 4.16.4, 4.17.0, 4.17.1, 4.17.2, 4.17.3, 4.18.0, 4.18.1, 4.18.2, 4.18.3, 4.19.0, 4.19.1, 4.19.2, 4.20.0, 4.21.0, 4.21.1
> @opentelemetry/instrumentation-fastify@0.40.0 test-all-versions
-- 1 package versions matching filter: 4.7.4
-- 1 package versions matching filter: 5.0.0
-- 11 package versions matching filter: 4.0.0, 4.24.3, 4.25.0, 4.25.1, 4.25.2, 4.26.0, 4.26.1, 4.26.2, 4.27.0, 4.28.0, 4.28.1
> @opentelemetry/instrumentation-graphql@0.43.0 test-all-versions
-- 18 package versions matching filter: 14.0.0, 14.5.8, 14.6.0, 15.3.0, 14.7.0, 15.4.0, 15.5.0, 15.5.1, 15.5.3, 15.6.0, 15.6.1, 15.7.2, 16.0.0, 15.8.0, 16.2.0, 16.3.0, 16.4.0, 15.9.0
> @opentelemetry/instrumentation-hapi@0.41.0 test-all-versions
-- 8 package versions matching filter: 21.3.3, 21.3.4, 21.3.5, 21.3.6, 21.3.7, 21.3.8, 21.3.9, 21.3.10
-- 1 package versions matching filter: 20.3.0
> @opentelemetry/instrumentation-ioredis@0.43.0 test-all-versions
-- 18 package versions matching filter: 2.5.0, 3.2.2, 4.14.1, 4.16.3, 4.17.3, 4.18.0, 4.19.2, 4.19.4, 4.22.0, 4.24.5, 4.26.0, 4.27.11, 4.28.5, 5.0.6, 5.2.6, 5.3.2, 5.4.0, 5.4.1
> @opentelemetry/instrumentation-koa@0.43.0 test-all-versions
-- 21 package versions matching filter: 2.7.0, 2.8.0, 2.8.1, 2.8.2, 2.9.0, 2.10.0, 2.11.0, 2.12.0, 2.12.1, 2.13.0, 2.13.1, 2.13.2, 2.13.3, 2.13.4, 2.14.0, 2.14.1, 2.14.2, 2.15.0, 2.15.1, 2.15.2, 2.15.3
-- 11 package versions matching filter: 8.0.8, 9.4.0, 10.0.0, 10.1.1, 11.0.1, 12.0.0, 12.0.1, 13.0.0, 13.0.1, 12.0.2, 13.1.0
> @opentelemetry/instrumentation-mongodb@0.47.0 test-all-versions
-- 14 package versions matching filter: 6.0.0, 6.1.0, 6.2.0, 6.3.0, 6.4.0, 6.5.0, 6.6.0, 6.6.1, 6.6.2, 6.7.0, 6.8.0, 6.8.1, 6.8.2, 6.9.0
-- 14 package versions matching filter: 5.0.0, 5.0.1, 5.1.0, 5.2.0, 5.3.0, 5.4.0, 5.5.0, 5.6.0, 5.7.0, 5.8.0, 5.8.1, 5.9.0, 5.9.1, 5.9.2
-- 31 package versions matching filter: 4.0.0, 4.0.1, 4.1.0, 4.1.1, 4.1.2, 4.1.3, 4.1.4, 4.2.0, 4.2.1, 4.2.2, 4.3.0, 4.3.1, 4.4.0, 4.4.1, 4.5.0, 4.6.0, 4.7.0, 4.8.0, 4.8.1, 4.9.0, 4.9.1, 4.10.0, 4.11.0, 4.12.0, 4.12.1, 4.13.0, 4.14.0, 4.15.0, 4.16.0, 4.17.0, 4.17.1
-- 38 package versions matching filter: 3.3.0, 3.3.1, 3.3.2, 3.3.3, 3.3.4, 3.3.5, 3.4.0, 3.4.1, 3.5.0, 3.5.1, 3.5.2, 3.5.3, 3.5.4, 3.5.5, 3.5.6, 3.5.7, 3.5.8, 3.5.9, 3.5.10, 3.6.0, 3.6.1, 3.5.11, 3.6.2, 3.6.3, 3.6.4, 3.6.5, 3.6.6, 3.6.7, 3.6.8, 3.6.9, 3.6.10, 3.6.11, 3.6.12, 3.7.0, 3.7.1, 3.7.2, 3.7.3, 3.7.4
> @opentelemetry/instrumentation-mysql2@0.41.0 test-all-versions
-- 5 package versions matching filter: 1.4.2, 1.5.3, 1.6.4, 1.6.5, 1.7.0
-- 5 package versions matching filter: 2.1.0, 2.2.5, 2.3.0, 2.3.2, 2.3.3
-- 49 package versions matching filter: 3.0.0, 3.1.0, 3.2.0, 3.2.1, 3.2.2, 3.2.3, 3.2.4, 3.3.0, 3.3.1, 3.3.2, 3.3.3, 3.3.4, 3.3.5, 3.4.0, 3.4.1, 3.4.2, 3.4.3, 3.4.4, 3.4.5, 3.5.0, 3.5.1, 3.5.2, 3.6.0, 3.6.1, 3.6.2, 3.6.3, 3.6.4, 3.6.5, 3.7.0, 3.7.1, 3.8.0, 3.9.0, 3.9.1, 3.9.2, 3.9.3, 3.9.4, 3.9.5, 3.9.6, 3.9.7, 3.9.8, 3.9.9, 3.10.0, 3.10.1, 3.10.2, 3.10.3, 3.11.0, 3.11.1, 3.11.2, 3.11.3
> @opentelemetry/instrumentation-nestjs-core@0.40.0 test-all-versions
-- 1 package versions matching filter: 6.11.11
-- 5 package versions matching filter: 7.6.12, 7.6.13, 7.6.15, 7.6.17, 7.6.18
-- 22 package versions matching filter: 8.0.6, 8.0.9, 8.0.11, 8.1.1, 8.1.2, 8.2.0, 8.2.1, 8.2.2, 8.2.3, 8.2.4, 8.2.5, 8.2.6, 8.3.0, 8.3.1, 8.4.0, 8.4.1, 8.4.2, 8.4.3, 8.4.4, 8.4.5, 8.4.6, 8.4.7
> @opentelemetry/instrumentation-pg@0.45.0 test-all-versions
-- 3 package versions matching filter: 8.5.1, 8.6.0, 8.7.1
> @opentelemetry/instrumentation-pino@0.42.0 test-all-versions
-- 15 package versions matching filter: 5.14.0, 5.17.0, 6.13.3, 7.2.0, 6.13.4, 7.8.0, 6.14.0, 7.11.0, 8.0.0, 8.4.0, 8.8.0, 8.12.1, 8.16.2, 8.20.0, 8.21.0
-- 7 package versions matching filter: 9.0.0, 9.1.0, 9.2.0, 9.3.0, 9.3.1, 9.3.2, 9.4.0
> @opentelemetry/instrumentation-redis@0.42.0 test-all-versions
-- 15 package versions matching filter: 2.6.0, 2.6.1, 2.6.2, 2.6.3, 2.6.4, 2.6.5, 2.7.0, 2.7.1, 2.8.0, 3.0.0, 3.0.1, 3.0.2, 3.1.0, 3.1.1, 3.1.2
> @opentelemetry/instrumentation-redis-4@0.42.1 test-all-versions
-- 13 package versions matching filter: 4.0.0, 4.0.6, 4.1.1, 4.2.0, 4.3.1, 4.4.0, 4.5.1, 4.6.11, 4.6.12, 4.6.13, 4.6.14, 4.6.15, 4.7.0
> @opentelemetry/instrumentation-restify@0.41.0 test-all-versions
-- 2 package versions matching filter: 10.0.0, 11.1.0
> @opentelemetry/instrumentation-winston@0.40.0 test-all-versions
-- 15 package versions matching filter: 2.0.0, 2.4.0, 3.3.1, 3.6.0, 2.4.7, 3.10.0, 3.11.0, 3.12.0, 3.12.1, 3.13.0, 3.13.1, 3.14.0, 3.14.1, 3.14.2, 3.15.0
```

</details>


<details>
<summary>after</summary>

```
> @opentelemetry/instrumentation-amqplib@0.42.0 test-all-versions
-- 6 package versions matching filter: 0.5.6, 0.6.0, 0.7.1, 0.8.0, 0.9.1, 0.10.4
> @opentelemetry/instrumentation-cucumber@0.9.0 test-all-versions
-- 10 package versions matching filter: 10.0.1, 10.1.0, 10.2.1, 10.3.2, 10.4.0, 10.5.1, 10.6.0, 10.7.0, 10.8.0, 10.9.0
-- 19 package versions matching filter: 8.0.0, 8.1.2, 8.2.2, 8.3.1, 8.4.0, 8.5.3, 8.6.0, 8.7.0, 8.8.0, 8.9.1, 8.10.0, 8.11.1, 9.0.1, 9.1.2, 9.2.0, 9.3.0, 9.4.0, 9.5.1, 9.6.0
> @opentelemetry/instrumentation-dataloader@0.12.0 test-all-versions
-- 3 package versions matching filter: 2.0.0, 2.1.0, 2.2.2
> @opentelemetry/instrumentation-lru-memoizer@0.40.0 test-all-versions
-- 15 package versions matching filter: 1.3.1, 1.4.0, 1.5.0, 1.6.0, 1.7.0, 1.8.0, 1.9.0, 1.10.1, 1.11.2, 1.12.0, 1.13.0, 2.0.1, 2.1.4, 2.2.0, 2.3.0
> @opentelemetry/instrumentation-mongoose@0.42.0 test-all-versions
-- 8 package versions matching filter: 8.0.4, 8.1.3, 8.2.4, 8.3.5, 8.4.5, 8.5.5, 8.6.4, 8.7.0
-- 9 package versions matching filter: 7.0.5, 7.1.2, 7.2.4, 7.3.4, 7.4.5, 7.5.4, 7.6.13, 7.7.0, 7.8.2
-- 19 package versions matching filter: 5.9.29, 5.10.19, 5.11.20, 5.12.15, 5.13.22, 6.0.15, 6.1.10, 6.2.11, 6.3.9, 6.4.7, 6.5.5, 6.6.7, 6.7.5, 6.8.4, 6.9.3, 6.10.5, 6.11.6, 6.12.9, 6.13.3
> @opentelemetry/instrumentation-socket.io@0.42.0 test-all-versions
-- 4 package versions matching filter: 4.5.4, 4.6.2, 4.7.5, 4.8.0
-- 1 package versions matching filter: 3.1.2
-- 5 package versions matching filter: 2.1.1, 2.2.0, 2.3.0, 2.4.1, 2.5.1
> @opentelemetry/instrumentation-tedious@0.14.0 test-all-versions
-- 1 package versions matching filter: 19.0.0
-- 1 package versions matching filter: 18.6.2
-- 1 package versions matching filter: 17.0.0
-- 5 package versions matching filter: 12.3.0, 13.2.0, 14.7.0, 15.1.3, 16.7.1
-- 11 package versions matching filter: 1.15.0, 2.7.1, 3.0.1, 4.2.0, 5.0.3, 6.7.1, 7.0.0, 8.3.1, 9.2.3, 10.0.0, 11.8.0
> @opentelemetry/instrumentation-undici@0.6.0 test-all-versions
-- 7 package versions matching filter: 6.0.0, 6.4.0, 6.9.0, 6.13.0, 6.18.1, 6.19.6, 6.19.8
-- 7 package versions matching filter: 5.12.0, 5.16.0, 5.21.0, 5.25.1, 5.26.3, 5.28.2, 5.28.4
> @opentelemetry/instrumentation-aws-sdk@0.44.0 test-all-versions
-- 7 package versions matching filter: 3.24.0, 3.128.0, 3.259.0, 3.370.0, 3.507.0, 3.665.0, 3.667.0
-- 7 package versions matching filter: 3.6.1, 3.88.0, 3.236.0, 3.6.4, 3.503.1, 3.665.0, 3.667.0
-- 7 package versions matching filter: 2.308.0, 2.584.0, 2.858.0, 2.1135.0, 2.1412.0, 2.1689.0, 2.1691.0
> @opentelemetry/instrumentation-bunyan@0.41.0 test-all-versions
-- 1 package versions matching filter: 1.8.15
> @opentelemetry/instrumentation-express@0.43.0 test-all-versions
-- 6 package versions matching filter: 4.16.4, 4.17.3, 4.18.3, 4.19.2, 4.20.0, 4.21.1
> @opentelemetry/instrumentation-fastify@0.40.0 test-all-versions
-- 1 package versions matching filter: 4.7.4
-- 1 package versions matching filter: 5.0.0
-- 7 package versions matching filter: 4.0.0, 4.5.1, 4.10.1, 4.17.0, 4.23.1, 4.27.0, 4.28.1
> @opentelemetry/instrumentation-graphql@0.43.0 test-all-versions
-- 7 package versions matching filter: 14.0.0, 14.4.0, 14.5.7, 15.5.1, 15.8.0, 16.8.2, 15.9.0
> @opentelemetry/instrumentation-hapi@0.41.0 test-all-versions
-- 1 package versions matching filter: 21.3.10
-- 1 package versions matching filter: 20.3.0
> @opentelemetry/instrumentation-ioredis@0.43.0 test-all-versions
-- 7 package versions matching filter: 2.0.0, 4.2.0, 4.11.0, 4.19.2, 4.27.9, 5.3.2, 5.4.1
> @opentelemetry/instrumentation-koa@0.43.0 test-all-versions
-- 9 package versions matching filter: 2.7.0, 2.8.2, 2.9.0, 2.10.0, 2.11.0, 2.12.1, 2.13.4, 2.14.2, 2.15.3
-- 12 package versions matching filter: 8.0.8, 9.0.1, 9.1.0, 9.2.0, 9.3.1, 9.4.0, 10.0.0, 10.1.1, 11.0.2, 12.0.2, 13.0.1, 13.1.0
> @opentelemetry/instrumentation-mongodb@0.47.0 test-all-versions
-- 7 package versions matching filter: 6.0.0, 6.1.0, 6.3.0, 6.6.0, 6.6.2, 6.8.1, 6.9.0
-- 7 package versions matching filter: 5.0.0, 5.0.1, 5.2.0, 5.5.0, 5.7.0, 5.9.0, 5.9.2
-- 7 package versions matching filter: 4.0.0, 4.1.2, 4.3.0, 4.7.0, 4.11.0, 4.16.0, 4.17.1
-- 7 package versions matching filter: 3.3.0, 3.4.0, 3.5.5, 3.6.1, 3.6.7, 3.7.2, 3.7.4
> @opentelemetry/instrumentation-mysql2@0.41.0 test-all-versions
-- 20 package versions matching filter: 1.4.2, 1.5.3, 1.6.6, 1.7.0, 2.0.2, 2.1.0, 2.2.5, 2.3.2, 3.0.1, 3.1.2, 3.2.4, 3.3.5, 3.4.5, 3.5.2, 3.6.5, 3.7.1, 3.8.0, 3.9.9, 3.10.3, 3.11.3
> @opentelemetry/instrumentation-nestjs-core@0.40.0 test-all-versions
-- 1 package versions matching filter: 6.11.11
-- 5 package versions matching filter: 7.6.12, 7.6.13, 7.6.15, 7.6.17, 7.6.18
-- 15 package versions matching filter: 8.0.11, 8.1.2, 8.2.6, 8.3.1, 8.4.7, 9.0.11, 9.1.6, 9.2.1, 9.3.12, 9.4.3, 10.0.5, 10.1.3, 10.2.10, 10.3.10, 10.4.4
> @opentelemetry/instrumentation-pg@0.45.0 test-all-versions
-- 7 package versions matching filter: 8.0.0, 8.2.0, 8.4.0, 8.7.0, 8.11.0, 8.11.6, 8.13.0
> @opentelemetry/instrumentation-pino@0.42.0 test-all-versions
-- 7 package versions matching filter: 5.14.0, 6.7.0, 7.1.0, 7.9.2, 8.12.0, 8.19.0, 8.21.0
-- 5 package versions matching filter: 9.0.0, 9.1.0, 9.2.0, 9.3.2, 9.4.0
> @opentelemetry/instrumentation-redis@0.42.0 test-all-versions
-- 5 package versions matching filter: 2.6.5, 2.7.1, 2.8.0, 3.0.2, 3.1.2
> @opentelemetry/instrumentation-redis-4@0.42.1 test-all-versions
-- 8 package versions matching filter: 4.0.6, 4.1.1, 4.2.0, 4.3.1, 4.4.0, 4.5.1, 4.6.15, 4.7.0
> @opentelemetry/instrumentation-restify@0.41.0 test-all-versions
-- 3 package versions matching filter: 10.0.0, 11.0.0, 11.1.0
> @opentelemetry/instrumentation-winston@0.40.0 test-all-versions
-- 7 package versions matching filter: 2.0.0, 2.4.0, 3.2.1, 3.5.0, 2.4.7, 3.14.1, 3.15.0
```

</details>

**Comment by trentm** [comment]
I don't recall how to force a TAV run on this PR.

**Comment by pichlermarc** [comment]
I'm sponsoring this.

File: plugins/node/opentelemetry-instrumentation-aws-sdk/.tav.yml [hunk 1]
```diff
@@ -1,19 +1,26 @@
 "aws-sdk":
   # A small subset of releases in the range [2.308.0, 3) to reduce testing time.
-  versions: "2.308.0 || 2.556.0 || 2.801.0 || 2.1049.0 || 2.1297.0 || 2.1546.0 || >=2.1548.0"
+  versions:
+    include: "^2.308.0"
+    mode: max-7
```
Comment (pichlermarc) [inline plugins/node/opentelemetry-instrumentation-aws-sdk/.tav.yml:5]: TIL that this mode exists. Nice. :+1:

**Review by pichlermarc — APPROVED** [review]

**Comment by pichlermarc** [comment]
> I don't recall how to force a TAV run on this PR.

@trentm I see in the ended up finding the way to do it (re-assigning a label). 

It's kind of a hack though, the workflow is not triggered directly when we apply labels, as we don't use the token of an actual github user, so the action is then performed by https://github.com/apps/github-actions instead. We could use the token from @opentelemetrybot, but then we'd need to give it triage rights on the repo - I think such a change would need to go though the https://github.com/open-telemetry/community/ repo first.

**Review by maryliag — APPROVED** [review]

**Review by pichlermarc — CHANGES_REQUESTED** [review]
Oh, looks like we're hitting some versions that are failing tests, I think we need to take care of these first before merging.

**Comment by trentm** [comment]
> Oh, looks like we're hitting some versions that are failing tests

Yup, I kind of expected some possible breakage -- I guess I should have started in draft. There were a few instrumentations that supported a version range larger than what TAV had been testing. I'll work through them.

**Comment by trentm** [comment]
Repeating the commit messages from the last 4 commits:

- instr-aws-sdk TAV tests: skip bad releases of client-sqs

    I believe `@aws-sdk/client-sqs` releases 3.363.0 to 3.377.0 were bad releases (fixed in 3.378.0).  https://github.com/open-telemetry/opentelemetry-js-contrib/issues/1828#issuecomment-1834276719 shows how `@aws-sdk/client-s3@3.377.0` was a bad release.

    Version 3.363.0 was when the `client-*` packages were moved to using "migrated `@smithy` packages" https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.363.0

    The comment above shows a particular failing for client-s3. I don't have a smoking gun for the client-sqs old versions, but the tests fail on those versions and it isn't worth digging into the reasons. Let's just skip those versions.

- instr-pg: drop setting IN_TAV which stopped being used in #1393

- instr-nestjs-core TAV testing: fix for @nestjs/core@9 and 10

    Stop installing some '@nestjs/*' deps that aren't necessary for testing with more recent major versions.

    Also turn of nestjs logging for testing to make the output much cleaner.

- instr-restify: bump min-supported from 4.0.0 to 4.1.0 because of some issue with its spdy dep that is not worth digging into

    Also fine-tune the restify version at which there was a slight behaviour in error handling that broke tests: behaviour change was in v8.2.0, not in v8.0.0.

**Comment by trentm** [comment]
A couple issues in the pg tests:

The instrumentation (or at least the tests) don't actually work with
`pg >=8.0.0 <8.0.3`. These are 4yo versions, not worth digging into the failures,
just bump the min version to pg@8.0.3.

```
  1) pg
       "before all" hook for "should return an instrumentation":
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/runner/work/opentelemetry-js-contrib/opentelemetry-js-contrib/plugins/node/opentelemetry-instrumentation-pg/test/pg.test.ts)
      at listOnTimeout (node:internal/timers:569:17)
      at processTimers (node:internal/timers:512:7)
```

Then there was a flakiness in pg-pool.test.ts:


```
  1) pg-pool
       "after all" hook in "pg-pool":
     Error: done() called multiple times in hook <pg-pool "after all" hook> of file /Users/trentm/tm/opentelemetry-js-contrib11/plugins/node/opentelemetry-instrumentation-pg/test/pg-pool.test.ts; in addition, done() received error: AssertionError [ERR_ASSERTION]: expected to have 1 used connection
      at process.emit (node:events:519:28)
      at processEmit (/Users/trentm/tm/opentelemetry-js-contrib11/node_modules/signal-exit/index.js:199:34)
      at process.emit (/Users/trentm/tm/opentelemetry-js-contrib11/node_modules/source-map-support/source-map-support.js:516:21)
      at process._fatalException (node:internal/process/execution:188:25)
      at throwUnhandledRejectionsMode (node:internal/process/promises:391:5)
      at processPromiseRejections (node:internal/process/promises:470:17)
      at processTicksAndRejections (node:internal/process/task_queues:96:32)
```

cause by this test:

```
'should generate `db.client.connection.count` and `db.client.connection.pending_requests` metrics'
```

It was an async test case that did not return a promise, so it could be running
after the test framework thought it was done.

**Comment by maryliag** [comment]
I was the one who create the test `should generate 'db.client.connection.count' and 'db.client.connection.pending_requests' metrics` , and saw it fail on some PRs, but having a hard time to fix this locally. If you have any suggestion, I'm happy to try it out and open a PR with the fix

**Comment by trentm** [comment]
> * I believe `@aws-sdk/client-sqs` releases 3.363.0 to 3.377.0 were bad releases (fixed in 3.378.0).

I believe we should avoid releases 3.363.0 to 3.377.0 for *all* `@aws-sdk/client-*`. The most recent test run failed on client-s3@3.363.0.

**Comment by trentm** [comment]
Now that the TAV run is completing, we can compare the compute-time usage differences.

**Before:**  13h 6m 31s

Screenshot 2024-10-11 at 2 58 32 PM

**After:** 4h 2m 43s

Screenshot 2024-10-11 at 2 59 55 PM

**Review by pichlermarc — APPROVED** [review]
amazing improvement, thanks for taking the time to work on this 🚀

**Pull request merged**